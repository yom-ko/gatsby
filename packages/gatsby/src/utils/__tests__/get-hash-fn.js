const getHashFn = require(`../get-hash-fn`)

describe(`Test hashing function`, () => {
  it(`default parameters`, () => {
    const hash = getHashFn({})(`my input`)
    expect(hash).toMatchSnapshot()
  })
  it(`guards against collisions`, () => {
    const hash = getHashFn({})(`my input`)
    try {
      getHashFn({ cache: new Set([hash]) })(`my input`)
    } catch (err) {
      expect(err).toMatchSnapshot()
    }
  })
})
