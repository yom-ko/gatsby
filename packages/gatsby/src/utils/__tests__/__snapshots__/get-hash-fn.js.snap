// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test hashing function default parameters 1`] = `174577032270956`;

exports[`Test hashing function guards against collisions 1`] = `[Error: Hash collision at f(my input) = 174577032270956]`;
